From 62f88c757a46751cd7b8f30a0dc8098c0b1e64bb Mon Sep 17 00:00:00 2001
From: Tim-cy Yang <Tim-cy.Yang@mediatek.com>
Date: Fri, 8 Nov 2024 11:36:26 +0800
Subject: [PATCH] mediatek-sdk: build host tool cryptsetup

Signed-off-by: Tim-cy Yang <Tim-cy.Yang@mediatek.com>
---
 package/Makefile           | 1 +
 package/libs/popt/Makefile | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/package/Makefile b/package/Makefile
index e93c4410a8..fd9aac7b93 100644
--- a/package/Makefile
+++ b/package/Makefile
@@ -65,6 +65,7 @@ $(curdir)/merge-index: $(curdir)/merge
 ifndef SDK
   $(curdir)/compile: $(curdir)/system/opkg/host/compile
 endif
+  $(curdir)/compile: $(curdir)/cryptsetup/host/compile
 
 $(curdir)/install: $(TMP_DIR)/.build $(curdir)/merge $(if $(CONFIG_TARGET_PER_DEVICE_ROOTFS),$(curdir)/merge-index)
 	- find $(STAGING_DIR_ROOT) -type d | $(XARGS) chmod 0755
diff --git a/package/libs/popt/Makefile b/package/libs/popt/Makefile
index ba9c82349b..93ce150e69 100644
--- a/package/libs/popt/Makefile
+++ b/package/libs/popt/Makefile
@@ -23,7 +23,10 @@ PKG_REMOVE_FILES:=autogen.sh aclocal.m4
 PKG_INSTALL:=1
 PKG_BUILD_PARALLEL:=1
 
+HOST_BUILD_PREFIX:=$(STAGING_DIR_HOST)
+
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/host-build.mk
 
 TARGET_CFLAGS += $(FPIC)
 
@@ -52,4 +55,5 @@ define Package/libpopt/install
 endef
 
 $(eval $(call BuildPackage,libpopt))
+$(eval $(call HostBuild))
 
-- 
2.45.2

